lib/background.js

```diff
@@ -2,14 +2,51 @@
 
 'use strict';
 
+/* borrowed from Adam's ChromeNoMore404s plugin */
+function wmAvailabilityCheck(url, onsuccess, onfail, onerror) {
+    var xhr = new XMLHttpRequest();
```

Comment:
So, what do you think about switching this to use the new [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) instead?  (It's quite a bit cleaner! :wink:)